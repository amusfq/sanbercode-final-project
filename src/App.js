import React, { useEffect } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom";
import Loadable from 'react-loadable';
import {Navbar} from './Components';

const About = Loadable({
  loader: () => import('./Pages/About'),
  loading: Loading
});
const Works = Loadable({
  loader: () => import('./Pages/Works'),
  loading: Loading
});
const Contact = Loadable({
  loader: () => import('./Pages/Contact'),
  loading: Loading
});
function App() {

  useEffect( () => {
    const elm = document.querySelector('#startingLoader');
    window.onload = () => {
      if (elm) {
        elm.remove();
      }
    }
  },[])

  return (
    <Router>
      <Navbar pos="top"/>
      <Switch>
        <Route exact path="/">
          <About/>
        </Route>
        <Route path="/works">
          <Works/>
        </Route>
        <Route path="/contact">
          <Contact/>
        </Route>
      </Switch>
      <Navbar pos="bottom"/>
    </Router>
  );
}

function Loading() {
  return (
    <div className="relative">
      <h1 className="text-2xl absolute font-bold" style={{top: '50%', left: '50%', transform: 'translate(-50%, -50%)'}}>Wait a secondâ€¦</h1>
    </div>
  );
}

export default App;
