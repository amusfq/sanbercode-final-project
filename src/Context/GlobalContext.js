import React, { createContext, useState } from 'react';

export const GlobalContext = createContext();

export const GlobalProvider = props => {
    const [activeNavbar, setActiveNavbar] = useState(0);
    const [theme, setTheme] = useState('white');
    const [dataWork, setDataWork] = useState([
        {
            judul: 'Website Portal',
            tipe: 'web',
            deskripsi: 'Portal Website Sekolah SMK YPM 1 Taman',
            link_gambar: 'https://amusfq.my.id/dist/img/ypm.jpg'
        },
        {
            judul: 'TOKOKU',
            tipe: 'web',
            deskripsi: 'Aplikasi POS Berbasis Web',
            link_gambar: 'https://amusfq.my.id/dist/img/pos-web.png'
        },
        {
            judul: 'SimPOS',
            tipe: 'desktop',
            deskripsi: 'Aplikasi POS Berbasis Desktop',
            link_gambar: 'https://amusfq.my.id/dist/img/pos-desktop.png'
        },
        {
            judul: 'MASS Calc',
            tipe: 'desktop',
            deskripsi: 'Aplikasi Penghitung Berat Badan Ideal berbasis Desktop',
            link_gambar: 'https://amusfq.my.id/dist/img/BMI.png'
        },
        {
            judul: 'CCE',
            tipe: 'desktop',
            deskripsi: 'Aplikasi Encryptor Caesar dan Vigenere berbasis Desktop',
            link_gambar: 'https://amusfq.my.id/dist/img/kriptografi.png'
        }
    ]);

    const values = {
        navbarState: [activeNavbar, setActiveNavbar],
        works: [dataWork, setDataWork],
        baseTheme: [theme, setTheme]
    }

    return (
        <GlobalContext.Provider value={values}>
            {props.children}
        </GlobalContext.Provider>
    )
}