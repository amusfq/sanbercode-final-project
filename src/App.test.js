import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import {GlobalProvider} from './Context/GlobalContext';

afterEach(cleanup);

test('Check About Page', () => {
  const {queryByText} = render(
    <GlobalProvider>
      <App />
    </GlobalProvider>
  );

  const elm = queryByText(/Achmad Musyaffa/i);
  expect(elm).toMatchSnapshot('ACHMAD MUSYAFFA TAUFIQI')
});

test('Check Works Page', () => {
  const {queryAllByText} = render(
    <GlobalProvider>
      <App />
    </GlobalProvider>
  );
  
  const btn = queryAllByText(/Works/i)[0]
  fireEvent.click(btn)
  const elm = queryAllByText(/My Works/i)[0];
  expect(elm).toMatchSnapshot('My Works')
});

test('Check Repository Page', () => {
  const {queryAllByText} = render(
    <GlobalProvider>
      <App />
    </GlobalProvider>
  );
  
  const btn = queryAllByText(/repository/i)[0]
  fireEvent.click(btn)
  const elm = queryAllByText(/Repository/i)[0];
  expect(elm).toMatchSnapshot('Repository')
});

test('Check Contact Page', () => {
  const {queryAllByText} = render(
    <GlobalProvider>
      <App />
    </GlobalProvider>
  );
  
  const btn = queryAllByText(/contact/i)[0]
  fireEvent.click(btn)
  const elm = queryAllByText(/Contact Me/i)[0];
  expect(elm).toMatchSnapshot('Contact Me')
});

test('Change Theme', () => {
  const {container, queryAllByTitle} = render(
    <GlobalProvider>
      <App />
    </GlobalProvider>
  );
  
  const btn = queryAllByTitle(/Change theme/i)[0]
  fireEvent.click(btn)
  expect(container).toMatchSnapshot();
});
