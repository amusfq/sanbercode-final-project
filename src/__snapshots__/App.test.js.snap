// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change Theme 1`] = `
<div>
  <nav
    class="w-full lg:p-4 border-t border-gray-500 lg:border-0 sticky top-0 hidden lg:block bg-white"
  >
    <ul
      class="flex lg:flex-wrap justify-center lg:justify-end space-x-4 px-0 lg:px-24"
    >
      <li
        class="w-1/4 lg:w-auto text-center"
      >
        <a
          class="block py-2 font-bold border-b-2 hover:border-blue-600 border-white"
          href="/"
        >
          <div
            class="block lg:hidden flex flex-col"
          >
            <i
              class="fas fa-home"
            />
            <span
              class="text-sm font-normal"
            >
              Home
            </span>
          </div>
          <span
            class="hidden lg:block"
          >
            About
          </span>
        </a>
      </li>
      <li
        class="w-1/4 lg:w-auto text-center"
      >
        <a
          class="block py-2 font-bold border-b-2 hover:border-blue-600 border-white"
          href="/works"
        >
          <div
            class="block lg:hidden flex flex-col"
          >
            <i
              class="fas fa-briefcase"
            />
            <span
              class="text-sm font-normal"
            >
              Works
            </span>
          </div>
          <span
            class="hidden lg:block"
          >
            Works
          </span>
        </a>
      </li>
      <li
        class="w-1/4 lg:w-auto text-center"
      >
        <a
          class="block py-2 font-bold border-b-2 hover:border-blue-600 border-white"
          href="/repository"
        >
          <div
            class="block lg:hidden flex flex-col"
          >
            <i
              class="fab fa-github"
            />
            <span
              class="text-sm font-normal"
            >
              Repo
            </span>
          </div>
          <span
            class="hidden lg:block"
          >
            Repository
          </span>
        </a>
      </li>
      <li
        class="w-1/4 lg:w-auto text-center"
      >
        <a
          class="block py-2 font-bold border-b-2 hover:border-blue-600 border-blue-600 border-white"
          href="/contact"
        >
          <div
            class="block lg:hidden flex flex-col"
          >
            <i
              class="fas fa-phone"
            />
            <span
              class="text-sm font-normal"
            >
              Contact
            </span>
          </div>
          <span
            class="hidden lg:block"
          >
            Contact
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <div
    class="w-full px-4 py-8 text-center"
  >
    <h1
      class="text-2xl lg:text-4xl font-bold mb-4"
    >
      Contact Me
    </h1>
    <div
      class="flex flex-wrap mt-10"
    >
      <div
        class="w-full lg:w-1/3 flex flex-col mb-4 lg:mb-0"
      >
        <i
          class="fas fa-envelope mr-2 text-2xl mb-2"
        />
        <p>
          amusfq@gmail.com
        </p>
      </div>
      <div
        class="w-full lg:w-1/3 flex flex-col mb-4 lg:mb-0"
      >
        <i
          class="fas fa-phone fa-flip-horizontal mr-2 text-2xl mb-2"
        />
         +62 89612890670
      </div>
      <div
        class="w-full lg:w-1/3 flex flex-col mb-4 lg:mb-0"
      >
        <i
          class="fas fa-map-marked-alt mr-2 text-2xl mb-2"
        />
         Sidoarjo, Jawa Timur, Indonesia
      </div>
    </div>
  </div>
  <nav
    class="w-full lg:p-4 border-t border-gray-500 lg:border-0 sticky bottom-0 block lg:hidden bg-white"
  >
    <ul
      class="flex lg:flex-wrap justify-center lg:justify-end space-x-4 px-0 lg:px-24"
    >
      <li
        class="w-1/4 lg:w-auto text-center"
      >
        <a
          class="block py-2 font-bold border-b-2 hover:border-blue-600 border-white"
          href="/"
        >
          <div
            class="block lg:hidden flex flex-col"
          >
            <i
              class="fas fa-home"
            />
            <span
              class="text-sm font-normal"
            >
              Home
            </span>
          </div>
          <span
            class="hidden lg:block"
          >
            About
          </span>
        </a>
      </li>
      <li
        class="w-1/4 lg:w-auto text-center"
      >
        <a
          class="block py-2 font-bold border-b-2 hover:border-blue-600 border-white"
          href="/works"
        >
          <div
            class="block lg:hidden flex flex-col"
          >
            <i
              class="fas fa-briefcase"
            />
            <span
              class="text-sm font-normal"
            >
              Works
            </span>
          </div>
          <span
            class="hidden lg:block"
          >
            Works
          </span>
        </a>
      </li>
      <li
        class="w-1/4 lg:w-auto text-center"
      >
        <a
          class="block py-2 font-bold border-b-2 hover:border-blue-600 border-white"
          href="/repository"
        >
          <div
            class="block lg:hidden flex flex-col"
          >
            <i
              class="fab fa-github"
            />
            <span
              class="text-sm font-normal"
            >
              Repo
            </span>
          </div>
          <span
            class="hidden lg:block"
          >
            Repository
          </span>
        </a>
      </li>
      <li
        class="w-1/4 lg:w-auto text-center"
      >
        <a
          class="block py-2 font-bold border-b-2 hover:border-blue-600 border-blue-600 border-white"
          href="/contact"
        >
          <div
            class="block lg:hidden flex flex-col"
          >
            <i
              class="fas fa-phone"
            />
            <span
              class="text-sm font-normal"
            >
              Contact
            </span>
          </div>
          <span
            class="hidden lg:block"
          >
            Contact
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <button
    class="fixed bottom-0 right-0 mb-10 mr-10 h-12 w-12 bg-blue-600 hover:bg-blue-700 rounded-full"
    title="Change theme"
  >
    <i
      class="fas fa-sun"
    />
  </button>
</div>
`;

exports[`Check About Page: ACHMAD MUSYAFFA TAUFIQI 1`] = `
<h1
  class="font-bold text-xl lg:text-4xl"
>
  ACHMAD MUSYAFFA TAUFIQI
</h1>
`;

exports[`Check Contact Page: Contact Me 1`] = `
<h1
  class="text-2xl lg:text-4xl font-bold mb-4"
>
  Contact Me
</h1>
`;

exports[`Check Repository Page: Repository 1`] = `
<span
  class="hidden lg:block"
>
  Repository
</span>
`;

exports[`Check Works Page: My Works 1`] = `
<h1
  class="text-2xl lg:text-4xl font-bold mb-4"
>
  My Works
</h1>
`;
